src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -34,49 +38,87 @@
 public class ModelManager implements Model {
     private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
 
-    // TODO: Remove the null values which are a placeholder due to the multiple constructors.
-    // Also will have to change the relevant attributes to final.
-    private AlfredStorage storage = null;
-    private AddressBook addressBook = null;
+
+
     private final UserPrefs userPrefs;
-    private FilteredList<Person> filteredPersons = null;
+    private AlfredStorage storage = null;
 
     // EntityLists
-    private ParticipantList participantList = new ParticipantList();
-    private TeamList teamList = new TeamList();
-    private MentorList mentorList = new MentorList();
+    private FilteredList<Participant> participantList;
+    private FilteredList<Team> teamList;
+    private FilteredList<Mentor> mentorList;
+
+    /**
+     * Returns a {@code ModelManager} with the data from {@code storage}'s address book and {@code userPrefs}. <br>
+     * The data from the sample address book will be used instead if {@code storage}'s address book is not found,
+     * or an empty address book will be used instead if errors occur when reading {@code storage}'s address book.
+     */
+    public static Model initModelManager(AlfredStorage alfredStorage, ReadOnlyUserPrefs userPrefs) throws AlfredException {
+        Optional<ParticipantList> optionalParticipantList;
+        Optional<MentorList> optionalMentorList;
+        Optional<TeamList> optionalTeamList;
```

Comment:
I think if storage returns an empty EntityList, then  we should auto initialize it using the constructor instead of leaving it as optional type

Reply:
These are just a temporary variable under the method initModelManager(), so I think its alright?